### ISSUE TITLE
Support 2015 Default annotation if the schema and table are missing from the url

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/chaise#442]

**Comment by jrchudy** [comment]
https://github.com/informatics-isi-edu/ermrest/blob/master/user-doc/annotation.md#2015-default , attaching the associated wiki page detailing the use case for this annotation.

**Comment by jrchudy** [comment]
@robes, for data-entry how do we want to handle this? @hongsudt, @howdyjessie, and I discussed it after the division meeting. 

If a user types in a url that is incomplete (`...\data-entry\` or `...\data-entry\#1` or so on), do we want to infer the table they want to go to by looking at the `Default` annotation for catalog, schema, and table? Or should we be sending the user back to the landing page where they can then click on a valid data-entry table?

One other though that came to mind was giving them a selector for the data that is missing. Say they navigate to `...\data-entry\`, should they get a catalog selector? Or do we want to show them a list of the available `...\<catalog>\<schema>:<table>` combinations that the user is allowed to edit (submit to) This goes for any amount of missing info. So if they navigate to `...\data-entry\#1`, should they get a schema selector? Or do we want to show them a list of available `\schema:table` combinations that they can navigate to. And just to exhaust the considered cases. Should they get a table selector if they navigate to `....\data-entry\#1\Zebrafish` or see a list again of the ones they are allowed to edit?

This is thinking about ACL permissions at the same time.

**Comment by robes** [comment]
I would favor one of two approaches here:
1. if they don't specify the `<schema>:<table>` then use the one specified in the Default annotation, or
2. just give an error page.

I would not do more than that, like giving a schema and table selector or redirecting back to a landing page. To me it just seems like this is more of an error to ever get to this page without it being a proper link.

I would be fine with this app just giving an error message to the user. But if we think we need to support Default annotation then I suppose we can do that.

**Comment by jrchudy** [comment]
I agree with the case of data-entry that we don't want to infer a table the user wants to input data for. I would assume a user is only trying to go to data-entry to input data for a specific catalog, schema, and table. For data-entry showing error messages should be enough to notify the user that they are trying to navigate to somewhere they didn't expect and it wouldn't be useful getting defaulted to a page they didn't even know about. (redirect might just confuse them more)

Added record and recordset tags to this because it also applies to them.

**Comment by hongsudt** [comment]
For the error message, upon clicking ok, it should direct me to the landing
page.  This error handling approach should be in the common module.

**Comment by karlcz** [comment]
Shouldn't we remove the data-entry tag from this since we've decided that data-entry won't support default schema nor table?

**Comment by jrchudy** [comment]
Looking into this issue for recordedit (data-entry) helped us to realize that recordedit won't support this default behavior. To handle the default behavior for recordedit, we decided to notify the user and redirect them to the landing page. This is done in PR #467. Removing the `data-entry` tag as this no longer pertains to it.

**Comment by hongsudt** [comment]
removed recordsset tag since it has its own issue already.

**Comment by hongsudt** [comment]
This issue depends on #537.

**Comment by jrchudy** [comment]
This is handled in PR #548.

**Comment by jrchudy** [comment]
PR #548 has been merged, closing this issue.
